internal/modelproxy/request.go

```diff
@@ -65,21 +67,89 @@ func (pr *proxyRequest) parseModel() error {
 	}
 
 	var err error
+	// TODO (samos123): Improve this to not read the entire body into memory for files.
 	pr.body, err = io.ReadAll(pr.r.Body)
```

Comment:
I think we should remove the assignment of `pr.body` here for the multipart form and only do this for JSON requests. In our case where we are rewriting the body for multipart forms we duplicating the entire request body in memory. Instead, for multipart form, I think it would be better to parse the form values from the HTTP request, then construct a buffer for what we want to rewrite the body as, and assign the bytes that underly that buffer finally to `pr.body`.

Reply:
Agree, I will tackle that in this PR